### PR TITLE
[CVSRP-4165 ]Employees First Furloughed Page 

### PR DESCRIPTION
**Pull request by william-hatzar**
image

### CODE REVIEW

File: app/views/EmployeeFirstFurloughedView.scala.html [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+    @formHelper(action = EmployeeFirstFurloughedController.onSubmit(), 'autoComplete -> "off") {
+
+      @h1(messages("employeeFirstFurloughed.heading"))
+
+       @p(Html(messages("employeeFirstFurloughed.p1")))
+
+       @p(Html(messages("employeeFirstFurloughed.p2")))
+
+        @errorSummary(form.errors, Some("date-value.day"))
```
Comment (peckoverHMRC) [inline app/views/EmployeeFirstFurloughedView.scala.html:41]: I think the error summary should go above the h1 unless the calculator works differently

File: conf/app.routes [hunk 2]
```diff
@@ -141,3 +141,6 @@ POST        /furlough-in-last-tax-year                       controllers.Furloug
 
 GET         /multiple-furlough-unsupported                   controllers.CalculationUnsupportedController.multipleFurloughUnsupported()
 GET         /start-date-within-lookback-unsupported          controllers.CalculationUnsupportedController.startDateWithinLookbackUnsupported()
+
+GET        /employeeFirstFurloughed                              controllers.EmployeeFirstFurloughedController.onPageLoad()
```
Comment (peckoverHMRC) [inline conf/app.routes:145]: urls are usually hyphenated. Might want to drop Georgie a message if there isn't one on the ticket/slides

File: conf/messages.en [hunk 3]
```diff
@@ -878,3 +878,14 @@ furloughInLastTaxYear.error.required = Select one of the options below
 calculationUnsupported.heading = This calculation cannot be supported at the moment
 calculationUnsupported.p1 = You can do this calculation manually.
 calculationUnsupported.startAgain = Start a new calculation
+
+employeeFirstFurloughed.title = When was this employee first furloughed?
+employeeFirstFurloughed.heading = When was this employee first furloughed?
+employeeFirstFurloughed.p1 = This is the date when the employee was first furloughed, on or after 1 November 2020.
+employeeFirstFurloughed.p2 = We need this information because you told us this employee has been on more than one period of furlough since 1 November 2020, and was not eligible for the Coronavirus Job Retention Scheme before this date.
+employeeFirstFurloughed.checkYourAnswersLabel = EmployeeFirstFurloughed
```
Comment (peckoverHMRC) [inline conf/messages.en:886]: this and a few of the error messages need updating

File: app/forms/EmployeeFirstFurloughedFormProvider.scala [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+import java.time.LocalDate
+
+import forms.mappings.Mappings
+import javax.inject.Inject
+import play.api.data.Form
+
+class EmployeeFirstFurloughedFormProvider @Inject() extends Mappings {
+
+  def apply(): Form[LocalDate] = Form("value" -> localDate(invalidKey = "employeeFirstFurloughed.error.invalid"))
```
Comment (peckoverHMRC) [inline app/forms/EmployeeFirstFurloughedFormProvider.scala:27]: Extra validation is needed either now or in another subtask

**Review by peckoverHMRC — CHANGES_REQUESTED** [review]

**Comment by MJCallahanPage** [comment]
Will be merged as part of CVSRP-4170

**Pull request closed without merging**